server/API/REST.php

```diff
@@ -590,6 +589,75 @@ function cancel_booking($vars) {
 	}
 }
 
+if(!function_exists('set_lecture_online_status')){
+	function set_lecture_online_status($vars){	
+		global $_PATCH;
+		$lectureId = intval($vars['lectureId']);
+		$status = $_PATCH['value'] == 'true' ? LECTURE_REMOTE : ~LECTURE_REMOTE;
+
+		try {
+			if (!isset($_SESSION['user_id'])) {
+				throw new ErrorException('Auth Needed');
+			}
+
+			$userId = intval($_SESSION['user_id']);
+			$pdo = new PDO('sqlite:../db.sqlite');
+
+			// Check user exists and is teacher
+			$stmt = $pdo->prepare('SELECT * FROM users WHERE ID = :userId AND type = :teacher');
+			$stmt->bindValue(':userId', $userId, PDO::PARAM_INT);
+			$stmt->bindValue(':teacher', USER_TYPE_TEACHER, PDO::PARAM_INT);
+			if (!$stmt->execute()) {
+				throw new PDOException($stmt->errorInfo()[2]);
+			}
+			if (!$stmt->fetch()) {
+				throw new PDOException('Teacher ' . $userId . ' not found.');
+			}
+
+			// Check lecture exists, is assigned to teacher and > 30m to start
+			$nextHour = new DateTime('now', new DateTimeZone('UTC'));
+			$nextHour->modify('+30 minutes'); // Check for timezones discrepancies
+			$stmt = $pdo->prepare('SELECT *
+								   FROM lectures L, courses C
+								   WHERE L.course_id = C.ID
+									   AND L.ID = :lectureId
+									   AND C.teacher_id = :userId
+									   AND L.start_ts > :nextHour');
+			$stmt->bindValue(':lectureId', $lectureId, PDO::PARAM_INT);
+			$stmt->bindValue(':userId', $userId, PDO::PARAM_INT);
+			$stmt->bindValue(':nextHour', $nextHour->getTimestamp(), PDO::PARAM_INT);
+			if (!$stmt->execute()) {
+				throw new PDOException($stmt->errorInfo()[2]);
+			}
+			if (!$stmt->fetch()) {
+				throw new PDOException('Lecture ' . $lectureId . ' not found.');
+			}
+
+			// Set lecture status
+			if($status == LECTURE_REMOTE){
+				$stmt = $pdo->prepare('UPDATE lectures
+								   SET settings = settings | :online
+								   WHERE ID = :lectureId');
+			}
+			else{
+				$stmt = $pdo->prepare('UPDATE lectures
+								   SET settings = settings & :online
+								   WHERE ID = :lectureId');
+
+			}
```

Comment:
Is this the same as `UPDATE lectures SET settings = 0 WHERE ID = :lectureId`? I have trouble understanding it

Reply:
no. If you like that, you clear all the settings bit. This way you only write the correct bit